phoenix-shared/src/commonMain/paymentsdb/fr.acinq.phoenix.db/PaymentsMetadata.sq

```diff
@@ -24,6 +31,14 @@ CREATE TABLE IF NOT EXISTS payments_metadata (
     PRIMARY KEY (type, id)
 );
 
+-- Provides a mapping: `local_channels.channel_id` => `incoming_payments.payment_hash`.
+CREATE TABLE IF NOT EXISTS channel_id_map (
+    channel_id BLOB NOT NULL PRIMARY KEY,
+    payment_hash BLOB NOT NULL
+);
```

Comment:
I think your idea is better. The problem was a lack of index to perform the search. But using `received_with_type` should drastically reduce the search space. And if we're also sorting by date, it's probably the first search result. I'm going to switch to this solution.

Reply:
This didn't work as expected, because `received_with_type` is almost always `MULTIPARTS_V1` :man_facepalming: So adding the index doesn't help, and we're essentially back to scanning the `incoming_payments` table.

So I came up with another solution:
* I added a new column: `received_with_new_channel`, which acts like a boolean
* I added an index that uses both `received_with_new_channel` & `created_at`
* And this allows me to scan only IncomingPayments with a NewChannel part

Since the payment we're looking for is almost always the most recent in this query, the query/search is expected to be quite fast in practice. And I also figured out a way to perform the deserialization one-row-at-a-time, which should also speed things up.

Note that, my implementation uses an SQLite feature called [partial indexes](https://sqlite.org/partialindex.html), which are available since SQLite v3.8. And according to the Android [docs](https://developer.android.com/reference/android/database/sqlite/package-summary), v3.8 is available since API 21. So according to [Wikipedia](https://en.wikipedia.org/wiki/Android_version_history), since Android v5.0 which was first released November 4, 2014.

Is this OK ?

And according to this [wiki](https://github.com/yapstudios/YapDatabase/wiki/SQLite-version-(bundled-with-OS)), we're safe on iOS.

Reply:
Done in [59fbf0f](https://github.com/ACINQ/phoenix/pull/316/commits/59fbf0f291b8da695a7430c308314437f2904f11)

Reply:
> Is this OK ?
Yes we are expecting Android 24+ i.e. SQLite 3.9.

The solution you found with a new `received_with_new_channel` column is fine with me, it's not ideal but it's a limitation of our DB model that serialises complex objects.